### PR TITLE
gprelations enhancements

### PR DESCRIPTION
**Pull request by mdsupport**
1. Provide literals for gprelations types.  
2. Add sequence number to order gprelations records.

Based on [SF Code Review #264](https://sourceforge.net/p/openemr/code-review/264/)

### CODE REVIEW

**Comment by bradymiller** [comment]
Since @sunsetsystems did the review on this related code on sourceforge in past ( https://sourceforge.net/p/openemr/code-review/264/ ), will see if he has time for this code review (if not, then I'll take it on).

File: library/gprelations.inc.php [hunk 1]
```diff
@@ -7,23 +7,19 @@
...
 // By convention we require that type1 must be less than or equal to type2.
 //
-// As of this writing (2009-11-11), only documents-to-pnotes relations are
-// used. However expansion is anticipated, as well as the opportunity to
-// obsolete the issue_encounter table.
+// Since expansion is anticipated, as well as the opportunity to obsolete the 
+// issue_encounter table, please document linkings used in standard codebase.
+
+function getGpRelKey ($strTab) {
+    $rs = sqlQuery('SELECT seq FROM list_options WHERE list_id=? AND option_id=? AND activity=?',
```
Comment (sunsetsystems) [inline library/gprelations.inc.php]: list_options.seq is only supposed to serve a cosmetic purpose, so the user can order the list in whatever sequence is most useful. You instead want to use title and/or option_value for associated values.

**Comment by sunsetsystems** [comment]
What is the reason for the new list? Note that lists are supposed to be of possible interest to users, so if that is not the case here then some other way should be found to make the gprelations table easier to work with.

Also, what is the practical benefit of the new seq2 column?

File: library/gprelations.inc.php (hunk 1 above)
Comment (mdsupport) [inline library/gprelations.inc.php]: Choice seq brings relations with different alias together.

**Comment by mdsupport** [comment]
Use of lists avoids having to maintain the library for every new relation end point.
Some of the relations we use require links to be presented in consistent order.  seq2 provides that option.

**Comment by sunsetsystems** [comment]
I think better to keep them in code. Adding lists as a development tool would confuse users and give them another way to screw things up. Unless I'm missing something you can't use a new relation without writing code anyway, right?

**Comment by mdsupport** [comment]
That would be true for current coding style.  As soon as some abstraction is brought in, this becomes restrictive.  We have general purpose background workflow processes that are configured for various relationships and use status of the list entry (must be present and active) to execute related steps.  One of the simpler example is if a lab result arrives, users with specific role get a reminder.

This approach has worked for us.  But based on your experience if this would cause confusion, you could consider [original commit](https://github.com/mdsupport/openemr/blob/4474cab919b687eaa407fe58edefbfd829692c35/library/gprelations.inc.php) if it adds any value at all.  Unfortunately it won't work for our current state.  Thx.

**Comment by sunsetsystems** [comment]
Consider doing a PR that adds value to the project as it exists here. Your example of notifying users in specific roles of lab results does sound like a nice feature, and in that case a list that supports it seems appropriate. 

Infrastructure projects also have value but need community buy-in.

**Comment by sunsetsystems** [comment]
As for the "original commit" I thought that was deemed OK? But if ti doesn't help anyone then might as well let it be.

**Comment by sunsetsystems** [comment]
Hmm I think something went wrong with yesterday's commits.

**Pull request closed without merging**